src/test/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesExampleTest.java

```diff
@@ -139,102 +146,281 @@ public void shouldDemonstrateInteractiveQueries() throws Exception {
-
-    assertThat(wordCountInstances, hasItem(
-        new HostStoreInfo("localhost", port, Sets.newHashSet("word-count", "windowed-word-count"))
-    ));
-
-    Properties consumerConfig = new Properties();
-    consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
-    consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
-    consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
-    consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, "wait-for-output-consumer");
-    consumerConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
-    IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(consumerConfig, WORD_COUNT_OUTPUT, inputValues.size());
-    IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(consumerConfig, WINDOWED_WORD_COUNT_OUTPUT, inputValues.size());
-
-    // Fetch all key-value pairs from the word-count store
-    final Invocation.Builder
-        allRequest =
-        client.target(baseUrl + "/keyvalues/word-count/all")
-            .request(MediaType.APPLICATION_JSON_TYPE);
-
-    final List<KeyValueBean>
-        allValues =
-        Arrays.asList(new KeyValueBean("all", 1L),
-            new KeyValueBean("hello", 2L),
-            new KeyValueBean("kafka", 2L),
-            new KeyValueBean("lead", 1L),
-            new KeyValueBean("streams", 3L),
-            new KeyValueBean("to", 1L),
-            new KeyValueBean("world", 3L));
-    final List<KeyValueBean>
-        all = fetchRangeOfValues(allRequest,
-        allValues);
-    assertThat(all, equalTo(allValues));
-
-    // Fetch a range of key-value pairs from the word-count store
-    final List<KeyValueBean> expectedRange = Arrays.asList(
-        new KeyValueBean("all", 1L),
-        new KeyValueBean("hello", 2L),
-        new KeyValueBean("kafka", 2L));
-
-    final Invocation.Builder
-        request =
-        client.target(baseUrl + "/keyvalues/word-count/range/all/kafka")
-            .request(MediaType.APPLICATION_JSON_TYPE);
-    final List<KeyValueBean>
-        range = fetchRangeOfValues(request, expectedRange);
-
-    assertThat(range, equalTo(expectedRange));
-
-    // Find the instance of the Kafka Streams application that would have the key hello
-    final HostStoreInfo
-        hostWithHelloKey =
-        client.target(baseUrl + "/instance/word-count/hello")
-            .request(MediaType.APPLICATION_JSON_TYPE).get(HostStoreInfo.class);
-
-    // Fetch the value for the key hello from the instance.
-    final KeyValueBean result = client.target("http://" + hostWithHelloKey.getHost() +
-        ":" + hostWithHelloKey.getPort() +
-        "/state/keyvalue/word-count/hello")
-        .request(MediaType.APPLICATION_JSON_TYPE).get(KeyValueBean.class);
-
-    assertThat(result, equalTo(new KeyValueBean("hello", 2L)));
+  
+    bindHostAndStartRestProxy(port, host);
+  
+    if (proxy != null) {
```

Comment:
Should we fail if `proxy == null` ?

Reply:
Yes, we can. Will do the required change @mjsax